sklearn/datasets/_arff_parser.py

```diff
@@ -196,10 +196,10 @@ def _io_to_generator(gzip_file):
         # read arff data with chunks
         columns_to_keep = [col for col in columns_names if col in columns_to_select]
         dfs = [first_df[columns_to_keep]]
-        for data in chunk_generator(arff_container["data"], chunksize):
-            dfs.append(
-                pd.DataFrame(data, columns=columns_names, copy=False)[columns_to_keep]
-            )
+        dfs.extend(
+            pd.DataFrame(data, columns=columns_names, copy=False)[columns_to_keep]
+            for data in chunk_generator(arff_container["data"], chunksize)
+        )
```

Comment:
most of the changes in this PR are insignificant. But in places like this one, it can actually add quite a bit of value.

Therefore I don't mind having these rules enabled. In reality, once this PR is merged, we won't have much of an issue with the rule itself, since it affects a tiny portion of the code we write, and very often for the better.